### PR TITLE
IGAPP-499: Improve no internet connection error

### PR DESCRIPTION
**Pull request by charludo**
This pull request belongs to an issue on our [bugtracker](https://issues.integreat-app.de/).
You can find it there by looking for an issue with the key which is mentioned in the title of this pull request.
It starts with the keyword **IGAPP**.

### CODE REVIEW

**Comment by charludo** [comment]
Some new translations might be needed (see comment on the issue tracker)

File: web/src/components/FailureSwitcher.tsx [hunk 1]
```diff
@@ -29,6 +29,7 @@ const FailureSwitcher = ({ error }: PropsType): ReactElement => {
   }, [error])
 
   const getFailureProps = (error: Error): { goToPath?: string; goToMessage?: string; errorMessage: string } => {
+    /* eslint-disable @typescript-eslint/no-unnecessary-condition */
     if (error instanceof NotFoundError && error.city !== undefined && error.language !== undefined) {
```
Comment (charludo) [inline web/src/components/FailureSwitcher.tsx:33]: I really don't like disabling linter errors, but I have no clue why this one happens in the first place:

`error.city` and `error.language` *can* be undefined, as per `NotFoundError`'s type. Linter complains that there's no overlap between the first, and each of the last two conditionals.

If someone knows what's up, I'd love to remove the disable-line...

Comment (SteffiStuffel) [reply web/src/components/FailureSwitcher.tsx:33]: You do not need this comment. Linting only fails when you run `yarn lint` in `/web`-subfolder, as the types are from the api-client. If you run linting in parent directory, everything works fine, pipeline should also succeed.

**Review by SteffiStuffel — APPROVED** [review]
Tested "no internet error" on android emulator. Looks very nice :tada:

File: web/src/components/FailureSwitcher.tsx (hunk 1 above)
Comment (charludo) [reply web/src/components/FailureSwitcher.tsx:33]: Ah, makes sense. Thanks.

**Review by steffenkleinle — APPROVED** [review]
Tested on android

**Pull request merged**